### PR TITLE
[BugFix]  fix array_agg function ignore null 

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

https://github.com/StarRocks/StarRocksTest/issues/1376
## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Review by satanson — DISMISSED** [review]

File: be/src/exprs/agg/nullable_aggregate.h [hunk 1]
```diff
@@ -636,7 +636,7 @@ class NullableAggregateFunctionUnary final
                     state_data.is_null = false;
                     this->nested_function->merge(ctx, data_column, state_data.mutable_nest_state(), i);
                 } else if constexpr (!IgnoreNull) {
-                    this->data(state_data).is_null = false;
+                    this->data(states[i] + state_offset).is_null = false;
```
Comment (stdpain) [inline be/src/exprs/agg/nullable_aggregate.h:639]:
```suggestion
                    state_data.is_null = false;
```

**Review by kangkaisen — DISMISSED** [review]

**Review by stdpain — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
BE Build Success

**Comment by mofeiatwork** [comment]
run all

**Comment by wanpengfei-git** [comment]
BE Build Success

**Comment by stdpain** [comment]
run starrocks_admit_test

**Comment by mofeiatwork** [comment]
@Mergifyio rebase

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Review by satanson — APPROVED** [review]

**Review by stdpain — APPROVED** [review]

**Pull request merged**